File: LeetCode/LeetAgain/NumJewelsInStonesSln.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace LeetCode.LeetAgain
{
    class NumJewelsInStonesSln : ISolution
    {
        public int NumJewelsInStones(string J, string S)
        {
            HashSet<char> hs = new HashSet<char>(J.ToCharArray());
            int count = 0;
            foreach (char c in S)
            {
                if (hs.Contains(c))
                    count++;
            }

            return count;
        }

        public void Execute()
        {
            throw new NotImplementedException();
        }
    }
}

File: LeetCodePython/numJewelsInStonesSln.py
class Solution:
    def numJewelsInStones(self, J, S):
        """
        :type J: str
        :type S: str
        :rtype: int
        """
        count = 0
        jewels = set(J)
        for s in S:
            if s in jewels:
                count = count + 1

        return count
